src/main/io/spektrum_vtx_control.c

```diff
@@ -0,0 +1,233 @@
+// Would need a PG item to survive power cycle. Not really used so let it be as is.
+uint8_t SpektrumRegion = SPEKTRUM_VTX_REGION_NONE;
+#endif
+
+// Mark an inital invalid VTX ctrl frame to force first VTX settings cheange to actually come from Tx/Rx.
+static uint32_t vtxControl_ipc = ~(SPEKTRUM_VTX_CONTROL_FRAME);
+
+// ############ RX task ######################
+void spektrumHandleVtxControl(uint32_t vtxCntrl)
+{
+  vtxControl_ipc = vtxCntrl;
+}
+// ###########################################
+
+
+// ############ VTX_CONTROL task #############
+void spektrumVtxControl(void)
+{
+    static uint32_t prevVtxControl =0;
+    uint32_t vtxControl;
+
+    // Check for invalid VTX ctrl frames
+    if ((vtxControl_ipc & SPEKTRUM_VTX_CONTROL_FRAME_MASK) != SPEKTRUM_VTX_CONTROL_FRAME) return;
+
+    vtxControl = vtxControl_ipc;
+    vtxControl_ipc = 0;
+
+    if (prevVtxControl == vtxControl) return;
+    prevVtxControl = vtxControl;
+
+    spektrumVtx_t vtx = {
+        .pitMode = (vtxControl & SPEKTRUM_VTX_PIT_MODE_MASK) >> SPEKTRUM_VTX_PIT_MODE_SHIFT,
+        .region  = (vtxControl & SPEKTRUM_VTX_REGION_MASK)   >> SPEKTRUM_VTX_REGION_SHIFT,
+        .power   = (vtxControl & SPEKTRUM_VTX_POWER_MASK)    >> SPEKTRUM_VTX_POWER_SHIFT,
+        .band    = (vtxControl & SPEKTRUM_VTX_BAND_MASK)     >> SPEKTRUM_VTX_BAND_SHIFT,
+        .channel = (vtxControl & SPEKTRUM_VTX_CHANNEL_MASK)  >> SPEKTRUM_VTX_CHANNEL_SHIFT,
+#ifdef USE_SPEKTRUM_REGION_CODES
+        .region  = (vtxControl & SPEKTRUM_VTX_REGION_MASK)   >> SPEKTRUM_VTX_REGION_SHIFT;
+#endif
+    };
+
+    const vtxSettingsConfig_t prevSettings = {
+        .band    = vtxSettingsConfig()->band,
+        .channel = vtxSettingsConfig()->channel,
+        .freq    = vtxSettingsConfig()->freq,
+        .power   = vtxSettingsConfig()->power,
+        .lowPowerDisarm = vtxSettingsConfig()->lowPowerDisarm,
+    };
+    vtxSettingsConfig_t newSettings = prevSettings;
+
+    vtxDevice_t *vtxDevice = vtxCommonDevice();
+    if (vtxDevice) {
+#ifdef USE_VTX_COMMON_FREQ_API
+        uint16_t freq = SpektrumVtxfrequencyTable[vtx.band][vtx.channel];
+        if (prevSettings.freq != freq) {
+            newSettings.band    = VTX_COMMON_BAND_USER;
+            newSettings.channel = vtx.channel;
+            newSettings.freq    = freq;
+        }
+#else
+        // Convert to the internal Common Band index
+        const uint8_t band    = spek2commonBand[vtx.band];
+        const uint8_t channel = vtx.channel +1; // 0 based to 1 based
+        if ((prevSettings.band != band) || (prevSettings.channel != channel)) {
+            newSettings.band    = band;
+            newSettings.channel = channel;
+            newSettings.freq    = vtxCommonLookupFrequency(vtxDevice, band, channel);
+        }
+#endif
+        // Seems to be no unified internal VTX API standard for power levels/indexes, VTX device brand specific.
+        const uint8_t power = convertSpektrumVtxPowerIndex(vtx.power);
```

Comment:
We don't have a fixed mapping between power level and actual power. It's also driver-independent. Please don't use any hardcoded Smartaudio/Tramp/whatever conversions. Use run-time conversion if possible instead so it will actually match the capabilities of the VTX connected.